controller/konnect/ops/ops_controlplane.go

```diff
@@ -113,7 +131,104 @@ func updateControlPlane(
 	}
 
 	cp.Status.SetKonnectID(*resp.ControlPlane.ID)
+
+	if err := setGroupMembers(ctx, cl, cp, sdkGroups); err != nil {
+		SetKonnectEntityProgrammedConditionFalse(cp, conditions.KonnectGatewayControlPlaneProgrammedReasonFailedToSetControlPlaneGroupMembers, err.Error())
+		return err
+	}
+
 	SetKonnectEntityProgrammedCondition(cp)
 
 	return nil
 }
+
+func setGroupMembers(
+	ctx context.Context,
+	cl client.Client,
+	cp *konnectv1alpha1.KonnectGatewayControlPlane,
+	sdkGroups ControlPlaneGroupSDK,
+) error {
+	if len(cp.Spec.Members) == 0 ||
+		cp.Spec.ClusterType == nil ||
+		*cp.Spec.ClusterType != sdkkonnectcomp.ClusterTypeClusterTypeControlPlaneGroup {
+		return nil
+	}
+
+	wg := sync.WaitGroup{}
+	wg.Add(len(cp.Spec.Members))
+	ch := make(chan string)
+	chErr := make(chan error)
+	errs := make([]error, 0)
+	for _, member := range cp.Spec.Members {
+		go func() {
+			var memberCP konnectv1alpha1.KonnectGatewayControlPlane
+			err := cl.Get(ctx, client.ObjectKey{Namespace: cp.Namespace, Name: member.Name}, &memberCP)
+			if err != nil {
+				if k8serrors.IsNotFound(err) {
+					chErr <- err
+					return
+				}
+				chErr <- fmt.Errorf("failed to get control plane group member %s: %w", member.Name, err)
```

Comment:
I believe this could be simplified as we do not change behavior when it is 404. 
```suggestion
				chErr <- fmt.Errorf("failed to get control plane group member %s: %w", member.Name, err)
```